### DISCUSSION TITLE
FastDDSGen autogenerate ROS2 Service scripts? [13738]

### DISCUSSION THREAD

**Discussion by wrgtri** [eProsima/Fast-DDS-Gen#236] [Ideas]
So I currently know you can use `fastddsgen -typeros2 -example CMake Message.idl` to generate a FastDDS publisher and subscriber that can then interact with ROS2 topics of the same message type.

However, I would like to do the same but with ROS2 services instead of ROS2 topics. In [this other repo](https://github.com/eProsima/Integration-Service/tree/main/examples/utils/dds/DDSAddTwoInts), there are files that are similar to the generated FastDDSGen files, but for services. Is there a command with FastDDSGen or another tool that will autogenerate these files that will allow me to interact with ROS2 services? (I would rather not use the Integration Service as I would like more direct communication as opposed to this added layer.)

**Comment by JLBuenoLopez** [comment]
Fast DDS-Gen is a tool to generate TypeSupports compatible with Fast DDS library. Its intention is not to provide support for ROS 2 environment. I am moving this to the Ideas forum where enhancements and feature requests are discussed, though it is not probable that this feature would be considered to be introduced into Fast DDS-Gen.
